lectures/5-Tutorials/SymmetricTensors.md

```diff
@@ -0,0 +1,1240 @@
+irreps to a spin-$1$ irrep, we can then convert it to a plain array and compare it to the
+$\mathrm{SU}(2)$ Clebsch-Gordan coefficients exported by the
+[WignerSymbols.jl package](https://github.com/Jutho/WignerSymbols.jl).
+
+```{code-cell} julia
+V1 = SU2Space(1 => 1)
+V2 = SU2Space(1//2 => 1)
+t = TensorMap(ones, ComplexF64, V1 ← V2 ⊗ V2)
+```
+
+```{code-cell} julia
+ta = convert(Array, t)
+```
+The conversion gives us a $3 \times 3 \times 2$ array, which exactly corresponds to the size
+of the $C_{1}^{\frac{1}{2},\frac{1}{2}}$ Clebsch-Gordan array. In order to explicitly
+compare whether the entries match we need to know the ordering of basis states assumed by
+TensorKit.jl when converting the tensor to its matrix elements in the irrep basis. For
+$\mathrm{SU}(2)$ the irrep basis is ordered in ascending magnetic quantum number $m$, which
+gives us a map $m = i - (l+1)$ for mapping an array index to a corresponding magnetic
+quantum number for the spin-$l$ irrep.
+```{code-cell} julia
+for i1 in 1:dim(V1), i2 in 1:dim(V2), i3 in 1:dim(V2)
+    # map basis state index to magnetic quantum number
+    m1 = i1 - (1 + 1)
+    m2 = i2 - (1//2 + 1)
+    m3 = i3 - (1//2 + 1)
+    @test ta[i1, i2, i3] ≈ clebschgordan(1//2, m2, 1//2, m3, 1, m1)
+end
+```
+
+Based on this discussion, we can quantify the aforementioned 'difficulties' in the inverse
+operation of what we just demonstrated, namely converting a given operator to a symmetric
+`TensorMap` given only its matrix elements in the irrep basis. Indeed, it is now clear that
+this precisely requires isolating the reduced matrix elements introduced above. Given the
+matrix elements of the operator in the irrep basis, this can in general be done by solving
+the system of equations implied by the [Clebsch-Gordan decomposition](wignereckart). A
+simpler way to achieve the same thing is to make use of the fact that the
+[Clebsch-Gordan tensors form a complete orthonormal basis](https://en.wikipedia.org/wiki/Clebsch%E2%80%93Gordan_coefficients#Orthogonality_relations)
+on the coupled space. Indeed, by projecting out the appropriate Clebsch-Gordan coefficients
+and using their orthogonality relations, we can construct a diagonal operator on each
+coupled irrep space $V^{(k)}$. Each of these diagonal operators is proportional to the
+identity, where the proportionality factor is precisely the reduced matrix element
+associated to the corresponding irrep fusion tree.
+```{figure} ../_static/SymmetricTensors/none2symm.svg
+:scale: 12%
+:name: none2symm
+```
+
+This procedure works for any group symmetry, and all we need are matrix elements of the
+operator in the irrep basis and the Clebsch-Gordan coefficients. In the following we
+demonstrate this explicit procedure for the particular example of $G = \mathrm{SU}(2)$.
+However, it should be noted that for general groups the Clebsch-Gordan coefficients may not
+be as easy to compute (if it is even known how). In addition, the procedure for manually
```

Comment:
```suggestion
be as easy to compute (in general, no closed formulas exist). In addition, the procedure for manually
```
I think in principle it is known how, at the very least for finite groups and lie groups, in the sense that you can come up with an algorithm that computes them, but not in an analytic way. I don't think this is something we need to go into, but this comment might be better